### ISSUE TITLE
[Nuxt 3 RC8] Vercel Crashes - Error 500 - `Cannot read properties of undefined (reading 'resolve')`

### ISSUE THREAD

**Issue by madebyfabian** [victorgarciaesgi/nuxt-typed-router#45]
First of all, thank you for this amazing package. I love typed routes and this was exactly what I needed. Hoping that this one will even be included by nuxt by default in the future.

I am using
- Nuxt 3, RC8
- 1.2.0 Version of this package

I have some issues though pushing to vercel when using this plugin. I do a normal `npm run build` on vercel, and the build logs say that everything was generated

![Bildschirmfoto 2022-09-05 um 17 02 04](https://user-images.githubusercontent.com/32813692/188477490-f1b9bf64-768d-4242-b2a3-dddd7574db0a.jpg)

Though I get this error:
```
2022-09-05T14:59:13.515Z	cf820afd-ef95-4c6c-abe9-10e053fdb3e0	ERROR	[nuxt] [request error] [unhandled] [500] Cannot read properties of undefined (reading 'resolve')
  at ./chunks/app/_nuxt/LayoutHeader.7bc606f8.mjs:22:31  
  at renderComponentSubTree (./chunks/handlers/renderer.mjs:10541:13)  
  at renderComponentVNode (./chunks/handlers/renderer.mjs:10476:16)  
  at Object.ssrRenderComponent (./chunks/handlers/renderer.mjs:10896:12)  
  at _sfc_ssrRender (./chunks/app/_nuxt/default.d7f3ff0e.mjs:40:32)  
  at renderComponentSubTree (./chunks/handlers/renderer.mjs:10541:13)  
  at renderComponentVNode (./chunks/handlers/renderer.mjs:10476:16)  
  at renderVNode (./chunks/handlers/renderer.mjs:10583:22)  
  at renderComponentSubTree (./chunks/handlers/renderer.mjs:10547:13)  
  at renderComponentVNode (./chunks/handlers/renderer.mjs:10476:16)
```

which basically tells me that my `$typedRouter.resolve` calls in the code are not available due to `$typedRouter` being undefined.
My config looks like:

```
nuxtTypedRouter: {
  outDir: './lib/router',
},
```

And I use both `$typedRouter` and `useTypedRouter`.

Did someone have this problem in the past and if so, how do you fix it? Is there something wrong with the config?

**Comment by victorgarciaesgi** [comment]
Hi! Thanks :D 
It's only with vercel? You don't have this problem when building in local env?

**Comment by madebyfabian** [comment]
@victorgarciaesgi Thanks for your response, yes, I am able to npm run build and then nuxt preview and the build works perfectly fine on local. So it seems that the module isn’t correctly being generated by vercel and therefore the global module is not registered/available I guess.

**Comment by madebyfabian** [comment]
I also have created a branch for the project where I removed everything from typed router and replaced it with the nuxt default router imlementation, and this works perfectly fine.

**Comment by victorgarciaesgi** [comment]
And you only have problems with `$typedRouter` and not `useTypeRouter`? I will look at it. Have you tried rc.9 ? Nuxt team is pushing a lot of breaking changes recently :p

**Comment by madebyfabian** [comment]
@victorgarciaesgi It seems like it's only with $typedRouter yes, but I cannot confirm a 100%. I have tried to install rc.9 but that basically broke every other module 😄 So quickly reverting back. But the teams are working on fixing all the modules for rc.9.

Thanks for looking into it! Let me know if there is anything I can help with.

**Comment by victorgarciaesgi** [comment]
There were problems breaking changes with modules in Nuxt > rc.6, I updated the package to super Nuxt rc.9, also modified the README. 
Your need to declare the module as a string and not a direct reference now! 
```ts
buildModules: [
   'nuxt-typed-router'
]
```

Can you try it?

**Comment by madebyfabian** [comment]
@victorgarciaesgi Hi there, so I already used it with the string reference, and not directly, so this surely wasn't the issue. I spinned up a fresh nuxt3 rc9 project (since I am still not able to upgrade mine due to incompat issues), and reployed it using the last version of nuxt-typed-router (1.2.0), and it works absolutely well.

Thanks again for looking into it! 

So it seems fixed when using rc9 + 1.2.0 (but unfortunately not rc8 + 1.2.0, this doesn't fix anything), I think the issue can be closed for now.
